sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -464,8 +472,14 @@ def unpause_snapshots(
                         snapshot.snapshot_id,
                         target_snapshot.snapshot_id,
                     )
+                    full_snapshot = snapshot.full_snapshot
```

Comment:
do you think it's worth hitting the cache here, or allowing full_snapshot to take in a cache?

Reply:
Maybe as a follow-up. It should happen rarely enough to not matter.